fusor-ember-cli/app/controllers/rhev.js

```diff
@@ -27,35 +29,68 @@ export default Ember.Controller.extend(NeedsDeploymentMixin, {
   hasEngine: Ember.computed.alias('deploymentController.hasEngine'),
   hasNoEngine: Ember.computed.not('hasEngine'),
 
-  hasHypervisor: Ember.computed('deploymentController.model.discovered_hosts.[]', function() {
-    return (this.get('deploymentController.model.discovered_hosts.length') > 0);
-  }),
+  hasHypervisor: Ember.computed(
+    'deploymentController.model.discovered_hosts.[]',
+    function() {
+      return (this.get('deploymentController.model.discovered_hosts.length') > 0);
+    }
+  ),
   hasNoHypervisor: Ember.computed.not('hasHypervisor'),
 
+  isEngineHostnameValid: Ember.computed.not('engineDiscoveredHostController.isHostnameInvalid'),
+  isHypervisorHostnameValid: Ember.computed.not('hypervisorDiscoveredHostController.isHostnameInvalid'),
+  
+
   disableTabRhevHypervisors: Ember.computed(
-    'deploymentController.model.rhev_is_self_hosted',
-    'hasNoEngine',
+    'isSelfHost',
+    'validRhevEngine',
     function() {
-      return (!this.get('deploymentController.model.rhev_is_self_hosted') && this.get('hasNoEngine'));
+      return (!this.get('isSelfHost') && !this.get('validRhevEngine'));
     }
   ),
 
   disableTabRhevConfiguration: Ember.computed(
-    'deploymentController.model.rhev_is_self_hosted',
-    'hasNoEngine',
-    'hasNoHypervisor',
+    'isSelfHost',
+    'validRhevEngine',
+    'validRhevHypervisor',
     function() {
-      return ((this.get('deploymentController.model.rhev_is_self_hosted') && this.get('hasNoEngine')) ||
-              (!this.get('deploymentController.model.rhev_is_self_hosted') && this.get('hasNoHypervisor'))
-             );
+      if (this.get('invalidRhevEngine')) {
```

Comment:
this is watching validRhevEngine but using invalidRhevEngine, which is a bit weird.  I think it could maybe have been:

```javascript
if (this.get('validRhevEngine')) {
    return !this.get('isSelfHost') && this.get('invalidRhevHypervisor');
}
return true;
```